Account/FirefoxAccount.swift

```diff
@@ -197,67 +198,52 @@ open class FirefoxAccount {
     }
     
     public class FxAProfile {
-        open var avatarUrl: URL?
-        open var avatarImageData: Data?
         open var displayName: String?
         open let email: String
+        open var avatar: Avatar?
         
         init(email: String, displayName: String?, avatar: String?) {
             self.email = email
             self.displayName = displayName
-            self.avatarImageData = nil
 
             if let avatarStr = avatar {
-                self.avatarUrl = URL(string: avatarStr)
-            } else {
-                self.avatarUrl = nil
-            }            
+                self.avatar = Avatar(url: URL(string: avatarStr))
+            }
         }
         
-        func downloadAvatarImage() {
-            guard let avatarUrl = self.avatarUrl else {
-                return
-            }
+        open class Avatar {
+            open var image: UIImage? = nil
+            open var downloadSucceded: Bool? = false
+            open var url: URL?
             
-            DispatchQueue.global().async {
-                self.avatarImageData = try? Data(contentsOf: avatarUrl)
-                NotificationCenter.default.post(name: NotificationFirefoxAccountProfileChanged, object: self)
+            init(url: URL?) {
+                guard let url = url else {
+                    self.image = UIImage(named: "placeholder-avatar")
+                    return
+                }
+                
+                SDWebImageManager.shared().downloadImage(with: url, options: SDWebImageOptions.continueInBackground, progress: nil) { (image, error, cacheType, success, url) in
```

Comment:
I think you probably want more options here.

http://cocoadocs.org/docsets/SDWebImage/4.1.0/Constants/SDWebImageOptions.html

Likely you want:

```
SDWebImageLowPriority
SDWebImageScaleDownLargeImages
```

Reply:
Looks like `SDWebImageScaleDownLargeImages` is not available for [3.7](http://cocoadocs.org/docsets/SDWebImage/3.7.0/Constants/SDWebImageOptions.html). Also, took a glance at the [4.0 migration](https://github.com/rs/SDWebImage/blob/master/Docs/SDWebImage-4.0-Migration-guide.md) guide and seems like they changed a lot which could make this big enough to be its own issue?